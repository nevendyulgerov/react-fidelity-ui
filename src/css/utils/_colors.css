.fidelity-ui-app {
  .color {
    &--primary {
      color: var(--color-primary) !important;
    }

    &--primary-light {
      color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      color: var(--color-primary--dark) !important;
    }

    &--secondary {
      color: var(--color-secondary) !important;
    }

    &--secondary-light {
      color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      color: var(--color-secondary--dark) !important;
    }

    &--success {
      color: var(--color-success) !important;
    }

    &--success-light {
      color: var(--color-success--light) !important;
    }

    &--success-dark {
      color: var(--color-success--dark) !important;
    }

    &--warning {
      color: var(--color-warning) !important;
    }

    &--warning-light {
      color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      color: var(--color-warning--dark) !important;
    }

    &--danger {
      color: var(--color-danger) !important;
    }

    &--danger-light {
      color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      color: var(--color-danger--dark) !important;
    }

    &--info {
      color: var(--color-info) !important;
    }

    &--info-light {
      color: var(--color-info--light) !important;
    }

    &--info-dark {
      color: var(--color-info--dark) !important;
    }

    &--body {
      color: var(--color-body) !important;
    }

    &--heading {
      color: var(--color-body--heading) !important;
    }

    &--body-bg {
      color: var(--color-body--bg) !important;
    }

    &--header {
      color: var(--header-color) !important;
    }

    &--header-bg {
      color: var(--header-color-bg) !important;
    }

    &--link {
      color: var(--color-body--link) !important;
    }

    &--facebook {
      color: var(--color-facebook) !important;
    }

    &--google {
      color: var(--color-google) !important;
    }

    &--white {
      color: var(--color-white) !important;
    }

    &--gray-1 {
      color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      color: var(--color-gray-5) !important;
    }

    &--gray-6 {
      color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      color: var(--color-gray-10) !important;
    }

    &--black {
      color: var(--color-black) !important;
    }
  }

  /* bg */
  .bg {
    &--primary {
      background-color: var(--color-primary) !important;
    }

    &--primary-light {
      background-color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      background-color: var(--color-primary--dark) !important;
    }

    &--secondary {
      background-color: var(--color-secondary) !important;
    }

    &--secondary-light {
      background-color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      background-color: var(--color-secondary--dark) !important;
    }

    &--success {
      background-color: var(--color-success) !important;
    }

    &--success-light {
      background-color: var(--color-success--light) !important;
    }

    &--success-dark {
      background-color: var(--color-success--dark) !important;
    }

    &--warning {
      background-color: var(--color-warning) !important;
    }

    &--warning-light {
      background-color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      background-color: var(--color-warning--dark) !important;
    }

    &--danger {
      background-color: var(--color-danger) !important;
    }

    &--danger-light {
      background-color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      background-color: var(--color-danger--dark) !important;
    }

    &--info {
      background-color: var(--color-info) !important;
    }

    &--info-light {
      background-color: var(--color-info--light) !important;
    }

    &--info-dark {
      background-color: var(--color-info--dark) !important;
    }

    &--facebook {
      background-color: var(--color-facebook) !important;
    }

    &--google {
      background-color: var(--color-google) !important;
    }

    &--body {
      background-color: var(--color-body) !important;
    }

    &--body--light {
      background-color: var(--color-body--light) !important;
    }

    &--heading {
      background-color: var(--color-body--heading) !important;
    }

    &--body-bg {
      background-color: var(--color-body--bg) !important;
    }

    &--header {
      background-color: var(--header-color) !important;
    }

    &--header-bg {
      background-color: var(--header-color-bg) !important;
    }

    &--link {
      background-color: var(--color-body--link) !important;
    }

    &--white {
      background-color: var(--color-white) !important;
    }

    &--gray-1 {
      background-color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      background-color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      background-color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      background-color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      background-color: var(--color-gray-5) !important;
    }

    &--cgray-6 {
      background-color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      background-color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      background-color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      background-color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      background-color: var(--color-gray-10) !important;
    }

    &--black {
      background-color: var(--color-black) !important;
    }

    &--transparent {
      background: transparent !important;
    }
  }

  /* border-color */
  .border-color {
    &--primary {
      border-color: var(--color-primary) !important;
    }

    &--primary-light {
      border-color: var(--color-primary--light) !important;
    }

    &--primary-dark {
      border-color: var(--color-primary--dark) !important;
    }

    &--secondary {
      border-color: var(--color-secondary) !important;
    }

    &--secondary-light {
      border-color: var(--color-secondary--light) !important;
    }

    &--secondary-dark {
      border-color: var(--color-secondary--dark) !important;
    }

    &--success {
      border-color: var(--color-success) !important;
    }

    &--success-light {
      border-color: var(--color-success--light) !important;
    }

    &--success-dark {
      border-color: var(--color-success--dark) !important;
    }

    &--warning {
      border-color: var(--color-warning) !important;
    }

    &--warning-light {
      border-color: var(--color-warning--light) !important;
    }

    &--warning-dark {
      border-color: var(--color-warning--dark) !important;
    }

    &--danger {
      border-color: var(--color-danger) !important;
    }

    &--danger-light {
      border-color: var(--color-danger--light) !important;
    }

    &--danger-dark {
      border-color: var(--color-danger--dark) !important;
    }

    &--info {
      border-color: var(--color-info) !important;
    }

    &--info-light {
      border-color: var(--color-info--light) !important;
    }

    &--info-dark {
      border-color: var(--color-info--dark) !important;
    }

    &--body {
      border-color: var(--color-body) !important;
    }

    &--body--light {
      border-color: var(--color-body--light) !important;
    }

    &--heading {
      border-color: var(--color-body--heading) !important;
    }

    &--body-bg {
      border-color: var(--color-body--bg) !important;
    }

    &--header {
      border-color: var(--header-color) !important;
    }

    &--header-bg {
      border-color: var(--header-color-bg) !important;
    }

    &--link {
      border-color: var(--color-body--link) !important;
    }

    &--facebook {
      border-color: var(--color-facebook) !important;
    }

    &--google {
      border-color: var(--color-google) !important;
    }

    &--white {
      border-color: var(--color-white) !important;
    }

    &--gray-1 {
      border-color: var(--color-gray-1) !important;
    }

    &--gray-2 {
      border-color: var(--color-gray-2) !important;
    }

    &--gray-3 {
      border-color: var(--color-gray-3) !important;
    }

    &--gray-4 {
      border-color: var(--color-gray-4) !important;
    }

    &--gray-5 {
      border-color: var(--color-gray-5) !important;
    }

    &--gray-6 {
      border-color: var(--color-gray-6) !important;
    }

    &--gray-7 {
      border-color: var(--color-gray-7) !important;
    }

    &--gray-8 {
      border-color: var(--color-gray-8) !important;
    }

    &--gray-9 {
      border-color: var(--color-gray-9) !important;
    }

    &--gray-10 {
      border-color: var(--color-gray-10) !important;
    }

    &--black {
      border-color: var(--color-black) !important;
    }
  }
}
