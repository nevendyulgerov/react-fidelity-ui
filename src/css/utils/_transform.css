.fidelity-ui-app {
  .rotate {
    transition: var(--transition);

    &--90 {
      transform: rotate(90deg) !important;
    }

    &--180 {
      transform: rotate(180deg) !important;
    }

    &--270 {
      transform: rotate(270deg) !important;
    }

    &--360 {
      transform: rotate(360deg) !important;
    }
  }
}
