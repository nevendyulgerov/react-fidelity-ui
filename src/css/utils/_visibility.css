.fidelity-ui-app {
  .visible {
    visibility: visible !important;
  }

  .invisible {
    visibility: hidden !important;
  }
}
