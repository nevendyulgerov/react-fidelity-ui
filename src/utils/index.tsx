/**
 * @description Is null
 * @param val
 */
export const isNull = (val: any) : boolean => val === null;

/**
 * @description Is array
 * @param val
 */
export const isArr = (val: any) : boolean => Array.isArray(val);

/**
 * @description Is object
 * @param val
 */
export const isObj = (val: any) : boolean => typeof val === 'object' && !isArr(val) && !isNull(val);

/**
 * @description Is number
 * @param val
 */
// eslint-disable-next-line no-restricted-globals
export const isNum = (val: any) : boolean => typeof val === 'number' && !isNaN(val);

/**
 * @description Is function
 * @param val
 */
export const isFunc = (val: any) : boolean => typeof val === 'function';

/**
 * @description Is string
 * @param val
 */
export const isStr = (val: any) : boolean => typeof val === 'string';

/**
 * @description Is undefined
 * @param val
 */
export const isUndef = (val: any) : boolean => typeof val === 'undefined';

/**
 * @description Is boolean
 * @param val
 */
export const isBool = (val: any) : boolean => typeof val === 'boolean';

/**
 * @description Is non empty string
 * @param val
 */
export const isNonEmptyStr = (val: any) : boolean => isStr(val) && val !== '';

/**
 * @description Uid
 * @param len
 */
export const uid = (len: number = 7): string => Math.random().toString(35).substr(2, len);

/**
 * @description Get object keys
 * @param obj
 * @returns {Array}
 */
export const getKeys = (obj: any) => Object.keys(obj);

/**
 * @description Each key
 * @param obj
 * @param callback
 */
export const eachKey = (obj: any, callback: (key: string, prop: any, index: number) => void) => {
  getKeys(obj).forEach((key, index) => callback(key, obj[key], index));
};

/**
 * @description Observe intersection
 * @param target
 * @param callback
 */
export const observeIntersection = (target: HTMLElement, callback: (items: []) => void) => {
  if (!isFunc(window.IntersectionObserver)) {
    return console.warn('IntersectionObserver is not supported.');
  }

  // eslint-disable-next-line compat/compat
  const Observe = new IntersectionObserver((items: []) => {
    callback(items);
  });

  Observe.observe(target);

  return Observe;
};

/**
 * @description Get css var
 * @param varName
 */
export const getCssVar = (varName: string): string | null => (
  getComputedStyle(document.documentElement).getPropertyValue(varName)
);

/**
 * @description Set css var
 * @param varName
 * @param varValue
 */
export const setCssVar = (varName: string, varValue: string | null) => {
  document.documentElement.style.setProperty(varName, varValue);
};

/**
 * @description Limit number
 * @param num
 * @param max
 */
export const limitNum = (num: number, max: number) => (num <= max ? num : max);

/**
 * @description Can render prop
 * @param prop
 */
export const isRenderable = (prop: any) => isNonEmptyStr(prop) || isNum(prop) || isObj(prop);
