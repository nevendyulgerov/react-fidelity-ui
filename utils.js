const fs = require('fs');

/**
 * @description Titlize
 * @param text
 * @param format
 */
const titlize = (text, format) => {
  const textWords = text.split('-');
  if (format === 'camel') {
    return textWords.map((word, index) => index > 0
      ? word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
      : word.charAt(0).toLowerCase() + word.slice(1).toUpperCase()).join('');
  } else if (format === 'pascal') {
    return textWords.map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()).join('');
  } else if (format === 'dashed') {
    return textWords.map(word => word.toLowerCase()).join('-');
  }
};

/**
 * @description createFile
 * @param componentName
 * @param path
 * @param fileExtension
 * @param content
 * @param fileName
 * @param baseDir
 */
const createFile = (componentName, path, fileExtension, content, fileName = 'index', baseDir = '') => {
  if (baseDir !== '' && !fs.existsSync(path)) {
    fs.mkdirSync(baseDir);
  }

  if (!fs.existsSync(path)) {
    fs.mkdirSync(path);
  }

  fs.writeFile(`${path}/${fileName}.${fileExtension}`, content, (err) => {
    if ( err ) {
      return console.log(err);
    }
  });
};

/**
 * @description createJavaScriptFile
 * @param componentName
 */
const createJavaScriptFile = componentName => {
  const componentPascalName = titlize(componentName, 'pascal');
  const componentDashedName = titlize(componentName, 'dashed');
  const baseDir = `./src/components/${componentPascalName}`;
  const componentPath = `./src/components/${componentPascalName}/script`;
  const content = `
import React from 'react';

const ${componentPascalName} = (props) => (
  <div data-component="${componentDashedName}">
  
  </div>
);

export default ${componentPascalName};

`.trim();

  createFile(componentName, componentPath, 'js', content, titlize(componentName, 'pascal'), baseDir);
};

/**
 * @description createJavaScriptIndexFile
 * @param componentName
 */
const createJavaScriptIndexFile = componentName => {
  const componentPascalName = titlize(componentName, 'pascal');
  const componentPath = `./src/components/${componentPascalName}/`;
  const content = `
import ${componentPascalName} from './script/${componentPascalName}';
export default ${componentPascalName}

`.trim();

  createFile('index', componentPath, 'js', content);
};

/**
 * @description createStyleFile
 * @param componentName
 */
const createStyleFile = componentName => {
  const componentPascalName = titlize(componentName, 'pascal');
  const componentDashedName = titlize(componentName, 'dashed');
  const componentPath = `./src/components/${componentPascalName}/style`;
  const content = `
/**
 * Styles for component ${componentPascalName}
 */

[data-component="${componentDashedName}"] {

}

`.trim();

  createFile(componentName, componentPath, "scss", content, componentPascalName);
};

exports.createComponent = componentName => {
  createJavaScriptFile(componentName);
  createJavaScriptIndexFile(componentName);
  createStyleFile(componentName);
};
