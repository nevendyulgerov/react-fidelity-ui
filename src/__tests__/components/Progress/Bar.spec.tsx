import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ProgressBar from '../../../components/Progress/Bar';
import { isObj } from '../../../utils';

describe('Components / Progress / Bar', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<ProgressBar value={25} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('attaches className "progress__bar" to wrapper', () => {
    const { baseElement } = render(<ProgressBar value={25} />);
    const progressBarNode = baseElement.querySelector('.progress__bar');

    expect(isObj(progressBarNode)).toBe(true);
  });

  it('assigns correct width', () => {
    const value = 25;
    const { baseElement } = render(<ProgressBar value={value} />);
    const progressBarNode = baseElement.querySelector('.progress__bar');

    // @ts-ignore
    expect(getComputedStyle(progressBarNode).width).toBe(`${value}%`);
  });

  it('attaches className matching "className" prop', () => {
    const className = 'my-selector';
    const { baseElement } = render(<ProgressBar value={25} className={className} />);
    const progressBarNode = baseElement.querySelector('.progress__bar');

    // @ts-ignore
    expect(progressBarNode.classList.contains(className)).toBe(true);
  });

  it('adds any prop to wrapper', () => {
    // test style prop
    const style = { height: '20px' };
    const { baseElement, rerender } = render(<ProgressBar value={25} style={style} />);
    const progressBarNode = baseElement.querySelector('.progress__bar');

    // @ts-ignore
    expect(getComputedStyle(progressBarNode).height).toBe(style.height);

    // test event/callback prop
    let isClickTriggered = false;
    const onClick = () => {
      isClickTriggered = true;
    };
    rerender(<ProgressBar value={25} onClick={onClick} />);

    // @ts-ignore
    fireEvent.click(progressBarNode);

    expect(isClickTriggered).toBe(true);

    // test attribute prop
    const role = 'progressbar';
    rerender(<ProgressBar value={25} role={role} />);

    // @ts-ignore
    expect(progressBarNode.getAttribute('role')).toBe(role);
  });
});
