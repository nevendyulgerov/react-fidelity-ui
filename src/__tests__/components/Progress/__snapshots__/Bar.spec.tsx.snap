// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Progress / Bar matches snapshot 1`] = `
<body>
  <div>
    <span
      class="progress__bar"
      role="progressbar"
      style="width: 25%;"
    />
  </div>
</body>
`;
